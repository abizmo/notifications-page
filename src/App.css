:root {
  --clr-primary: hsl(1, 90%, 64%);
  --clr-secondary: hsl(219, 85%, 26%);
  --clr-neutral-900: hsl(0, 0%, 100%);
  --clr-neutral-700: hsl(210, 60%, 98%);
  --clr-neutral-600: hsl(211, 68%, 94%);
  --clr-neutral-500: hsl(205, 33%, 90%);
  --clr-neutral-300: hsl(219, 14%, 63%);
  --clr-neutral-200: hsl(219, 12%, 42%);
  --clr-neutral-100: hsl(224, 21%, 14%);

  --sz-0: 0.25rem;
  --sz-1: 0.5rem;
  --sz-2: 0.75rem;
  --sz-3: 1rem;
  --sz-4: 1.25rem;
  --sz-5: 1.5rem;
  --sz-6: 1.75rem;
  --sz-7: 2rem;
  --sz-8: 2.25rem;
  --sz-9: 2.5rem;
  --sz-10: 2.75rem;
  --sz-11: 3rem;
  --sz-12: 3.25rem;
  --sz-13: 3.5rem;
  --sz-14: 3.75rem;
  --sz-15: 4rem;

  --br-900: 1rem;
  --br-700: 0.5rem;
  --br-500: 0.25rem;

  --fs-800: 1.5rem;
  --fs-700: 1.25rem;
  --fs-500: 1rem;
  --fs-400: 0.9375rem;
  --fs-300: 0.875rem;
  --fs-100: 0.6875rem;

  --fw-regular: 500;
  --fw-bold: 800;

  --lh-1: 1.1; /* h1 */
  --lh-2: 1.2; /* message */
  --lh-3: 1.3; /* notification */
  --lh-5: 1.5; /* counter, caption */

  --ls-300: -0.02em;
  --ls-200: -0.032em;
  --ls-100: -0.04em;

  --ff-sans: 'Plus Jakarta Sans', sans-serif;
}

body {
  background-color: var(--clr-neutral-600);
  color: var(--clr-neutral-100);

  font-family: var(--ff-sans);
  font-size: var(--fs-500);
  line-height: var(--lh-3);
}
