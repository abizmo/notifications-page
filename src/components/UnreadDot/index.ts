export { default as UnreadDot } from './UnreadDot';
