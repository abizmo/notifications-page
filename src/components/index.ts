export * from './Image';
export * from './UnreadDot';
