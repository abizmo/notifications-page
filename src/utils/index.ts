export { default as getTimeAgo } from './getTimeAgo';
