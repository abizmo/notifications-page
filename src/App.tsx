import React from 'react';

import { NotificationsPage } from './pages';
import './App.css';

function App() {
  return <NotificationsPage />;
}

export default App;
