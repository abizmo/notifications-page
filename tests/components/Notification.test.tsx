import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, it, vi } from 'vitest';

import { Notification } from '../../src/components';

describe('Notificaton', () => {
  afterEach(cleanup);
  const date = '2022-11-06T18:20:17.659Z';
  const user = {
    name: 'Mark Webber',
    avatar: 'images/avatar-mark-webber.webp',
  };
  vi.mock('../../src/utils', () => {
    return {
      getTimeAgo: vi.fn().mockReturnValue('8 Days'),
    };
  });
  const subject = 'Hello, World!';

  it('should render and matchSnapshot', () => {
    const { container } = render(
      <Notification date={date} user={user} unread={true}>
        <Notification.Subject>{subject}</Notification.Subject>
      </Notification>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should show username, subject, date and unreadDot', () => {
    render(
      <Notification date={date} user={user} unread={true}>
        <Notification.Subject>{subject}</Notification.Subject>
      </Notification>,
    );

    expect(screen.getByText(user.name)).toBeDefined();
    expect(screen.getByText(subject)).toBeDefined();
    expect(screen.getByText('8 Days')).toBeDefined();
    expect(screen.getByAltText(user.name)).toBeDefined();
    expect(screen.getByRole('status')).toBeDefined();
  });

  it('should show username, subject, date and no unreadDot', () => {
    render(
      <Notification date={date} user={user} unread={false}>
        <Notification.Subject>{subject}</Notification.Subject>
      </Notification>,
    );

    expect(screen.queryByRole('status')).not.toBeInTheDocument();
  });

  it('should show the link', () => {
    render(
      <Notification date={date} user={user} unread={false}>
        <Notification.Subject>
          {subject}
          <Notification.Link href='#'>Test Link</Notification.Link>
        </Notification.Subject>
      </Notification>,
    );

    expect(screen.queryByText(/test link/i)).toBeInTheDocument();
  });

  it('should show the message', () => {
    render(
      <Notification date={date} user={user} unread={false}>
        <Notification.Subject>{subject}</Notification.Subject>
        <Notification.Message href='#'>Test Message</Notification.Message>
      </Notification>,
    );

    expect(screen.queryByText(/test message/i)).toBeInTheDocument();
  });

  it('should show the picture', () => {
    render(
      <Notification date={date} user={user} unread={false}>
        <Notification.Group>
          <Notification.Subject>{subject}</Notification.Subject>
          <Notification.Picture alt='Test Picture' href='#' src='#' />
        </Notification.Group>
      </Notification>,
    );

    expect(screen.queryByAltText(/test picture/i)).toBeInTheDocument();
  });
});
