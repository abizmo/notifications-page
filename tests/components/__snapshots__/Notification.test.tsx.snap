// Vitest Snapshot v1

exports[`Notificaton > should render and matchSnapshot 1`] = `
<div>
  <div
    class="notification notificationUnread"
  >
    <div
      class="avatar"
    >
      <div
        class="container circle"
      >
        <img
          alt="Mark Webber"
          src="images/avatar-mark-webber.webp"
        />
      </div>
    </div>
    <div
      class="notificationBody"
    >
      <div
        class="notificationSubject"
      >
        <p>
          <span
            class="notificationUser"
          >
            Mark Webber
          </span>
           
          Hello, World!
          <span
            class="unreadDot"
            role="status"
          />
        </p>
        <p
          class="notificationTimeAgo"
        >
          8 Days
        </p>
      </div>
    </div>
  </div>
</div>
`;
